### ISSUE TITLE
Block with Full HTML format deletes link around image tag

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#1601]
email bug report from @maxmeno 
I need a sponsor-banner on top of every page. Created custom block, full HTML format, uploaded the logo jpg file, added a link, saved. Banner is there, link is gone, not saved. Why?

**Comment by maxmeno** [comment]
After more head banging, I realized that the problem is different: it seems that any editing on a custom block done after opening it from the page (through the wheel on top-right corner) or from the layout page **is not saved**. 
If I open the same block from the custom blocks list, I can add a link or anything else, save and it stays.

**Comment by jenlampton** [comment]
That does sound like a bug, though I haven't had the same problem. Can you reproduce it on more than one site? And on the latest version of Backdrop?

**Comment by maxmeno** [comment]
I currently have one functioning BD site and it's fully up-to-date.

Maybe I nailed the problem, anyway. Can anybody reproduce it this way?

A) Editing does not get saved** if I work on the HTML source**, after opening the block through the top right wheel on the page and/or from the layout page. Editing gets saved if I open the block the same way but then work on WYSIWYG.

B) Editing is always saved, no matter how I work, if the block has been opened from the custom blocks list.

Tested several times and on different blocks.
